packages/samples/headless-react/src/App.tsx

```diff
@@ -64,6 +76,14 @@ const resultsPerPage = buildResultsPerPage(engine, {
 const pager = buildPager(engine, {options: {numberOfPages: 6}});
 
 function App() {
+  useEffect(
+    () =>
+      searchParameterManager.subscribe(() =>
+        writeSearchParametersToURI(searchParameterManager.state.parameters)
+      ),
+    []
+  );
```

Comment:
Could we place the subscription in the same file as the serializer? The way these samples will be displayed is by associating a controller with one or more files containing samples. If one of the files is App.tsx though, then readers will see a large file and not know where to look?

If we could show the interaction in single file, or two focused files, the sample will be much clearer.